src/app/qgsmaptooltrimextendfeature.cpp

```diff
@@ -0,0 +1,279 @@
+/***************************************************************************
+    qgmaptooltrimextendfeature.cpp  -  map tool to trim or extend feature
+    ---------------------
+    begin                : October 2018
+    copyright            : (C) 2018 by Loïc Bartoletti
+    email                : loic dot bartoletti at oslandia dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 3 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsmaptooltrimextendfeature.h"
+
+#include "qgsmapcanvas.h"
+#include "qgsvertexmarker.h"
+#include "qgsvectorlayer.h"
+#include "qgsgeometry.h"
+#include "qgssnappingutils.h"
+#include "qgstolerance.h"
+#include "qgisapp.h"
+#include "qgsgeometryutils.h"
+#include "qgsmapmouseevent.h"
+#include "qgssnapindicator.h"
+
+class QgsRubberBand;
+
+class FeatureFilter : public QgsPointLocator::MatchFilter
+{
+  public:
+    FeatureFilter()
+    {}
+
+    bool acceptMatch( const QgsPointLocator::Match &match ) override
+    {
+      if ( mLayer )
+        return match.layer() == mLayer && match.hasEdge();
+
+      return match.hasEdge();
+    }
+    // We only want to modify the current layer. When geometries are overlapped, this makes it possible to snap onto the current layer.
+    void setLayer( QgsVectorLayer *vlayer ) { mLayer = vlayer; }
```

Comment:
small thing: use `layer` or `vl` for coherence, even if it is not part of API

Reply:
Done with `layer`